tests/kolainst/destructive/mount-propagation.sh

```diff
@@ -0,0 +1,127 @@
+#!/bin/bash
+# https://bugzilla.redhat.com/show_bug.cgi?id=1498281
+set -xeuo pipefail
+
+. ${KOLA_EXT_DATA}/libinsttest.sh
+
+require_writable_sysroot
+
+has_mount() {
+  local target=${1:?No target specified}
+  local pid=${2:?No PID specified}
+  local mounts="/proc/${pid}/mounts"
+
+  # Unlike grep, awk doesn't exit based on whether it matched anything or not,
+  # so that needs to be emulated.
+  awk 'BEGIN {ret = 1}; $2 == "'"${target}"'" {print; ret = 0}; END {exit ret}' "${mounts}"
```

Comment:
We could instead parse JSON output from e.g. `findmnt -J --task`...

Reply:
Yes, totally fair. The unfortunate part there is `findmnt` always reads `/proc/self/mountinfo` whereas I'm trying to read the mounts of an arbitrary process. You can do the same with `nsenter -m -t $pid findmnt`, but it's a bit more verbose.

Reply:
I changed this to use `nsenter findmnt --json | jq`. It's wordier but a little nicer. I think I'm in the minority of understanding `awk` :smile: